### PR TITLE
sema: diagnose @_spi_available on all platforms

### PR DESCRIPTION
**Pull request by nkcsgexi**
@_spi_available should only be used when the symbol is publically available in some deployment platforms, otherwise, we should use @_spi.

### CODE REVIEW

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

File: lib/Sema/TypeCheckAttr.cpp [hunk 1]
```diff
@@ -1421,8 +1422,9 @@ void TypeChecker::checkDeclAttributes(Decl *D) {
     TypeChecker::applyAccessNote(VD);
 
   AttributeChecker Checker(D);
-  // We need to check all OriginallyDefinedInAttr and BackDeployAttr relative
-  // to each other, so collect them and check in batch later.
+  // We need to check all availableAttrs, OriginallyDefinedInAttr and
+  // BackDeployAttr relative to each other, so collect them and check in batch later.
```
Comment (tshortli) [inline lib/Sema/TypeCheckAttr.cpp:1426]: Nit: `clang-format` for line length consistency.

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Comment by tshortli** [comment]
I think this diagnostic overall makes sense, and perhaps this is an edge case, but a missing availability annotation for a specific platform is expected in some circumstances, like for the standard library or any other library that is available at the start of a platform. Therefore enumerating all explicit annotations doesn't really answer the question of whether a declarations is SPI available on all platforms relevant to the module, since declarations are implicitly always available on platforms without annotations. So this diagnostic may be misleading in a small set of scenarios.
